src/color/color.js

```diff
@@ -0,0 +1,39 @@
+const flatten = require('../utils/flatten')
+
+const colorNameToRgb = require('./colorNameToRgb')
+
+/**
+ * Apply the given color to the given objects.
+ * @param {Object} color - either an array or a hex string of color values
+ * @param {Object|Array} objects - the objects of which to color
+ * @returns {Object|Array} the same objects with an additional attribute 'color'
+ *
+ * @example
+ * let redSphere = color([1,0,0,1], sphere())
```

Comment:
so the alpha for rgba is merged into the color array right ?
the only problem with that is for 'named/css' colors: they are rgb data, and without a seperate alpha parameter, we need to account for that somehow (hence some of the weirdness of the previous color op)

Reply:
one possible solution, is to not do any 'magic' in here:
if you want a specific color, you (as a user) should create your colors using hexToRgba or colorNameToRgb etc
it has one extra perk : less code and conditionals here, and it means each functions does one thing only but one thing well :)

Reply:
I was thinking very similar. 

I think the color() function can just set color. And like you mentioned, the other functions can be used to translate between specifications or names.

But, I think most people will not provide the ALPHA.

We can have the color() function provide an ALPHA if not (which the current code does)

OR

Like you said, let the viewer interpret colors if there’s no ALPHA.

Probably, we should do both.

Reply:
agreed, default to 1, but give the option to provide it (do  not give the option to set it in the color() function, but in the 'creator'/ converter function) :)